### ISSUE TITLE
Special character in hosts file is causing dnsmasq to malfunction

### ISSUE THREAD

**Issue by drakofrost** [StevenBlack/hosts#251]
There's an entry in the hosts file with a special character, that's preventing dnsmasq from working, and therefore, Pi-Hole.

lifehacĸer.com

The "ĸ" is not an actual "k", is a special character.

"systemctl status dnsmasq" returns this error:

bad name at /etc/pihole/gravity.list line 74173

Which is this line:

lifehacĸer.com

Removing the offending simbol from the file solves the issue and make dnsmasq work again. The problem is that when Pi-Hole updates it comes back. Can you remove this entry from the hosts file?

**Comment by StevenBlack** [comment]
This is an obvious phish, and we'll be seeing lots more of that, I fear.

Is the Pi-hole project aware of this problem?  https://github.com/pi-hole/pi-hole/issues

**Comment by drakofrost** [comment]
I was going to report to them, but since they use your list, they can't really fix the problem. Also, the issue is not in Pi-Hole, it seems that dnsmasq doesn't accept special characters.

**Comment by mitchellkrogza** [comment]
This is because special characters need to be converted to a string that your DNS or web server can understand. Luckily Nginx already does this for me.

Take for instance ```xn--oogle-wmc.com```

Put that in a browser and see what it changes to it says "Google.com" but it's not.
![screen shot 2017-02-23 at 9 21 02 am](https://cloud.githubusercontent.com/assets/9961541/23248727/94e37b62-f9a9-11e6-9602-2333f9892085.png)

Have a look at the list of converted domains on my list that any DNS can understand.
https://github.com/StevenBlack/hosts/pull/248

You can nslookup on any of those funny domain strings and get an answer on them.

**Comment by drakofrost** [comment]
When I paste the address in the browser nothing happens.

If I understood correctly, it's possible to avoid the issue by converting the string? If that's the case, it's something that Steven could do?

**Comment by mitchellkrogza** [comment]
Weird, I can do nslookups on those funny strings from any linux or Mac OS I've tested on or paste them into any of my browsers and it resolves.

**Comment by mitchellkrogza** [comment]
lifehacĸer.com translates to xn--lifehacer-1rb.com

```nslookup xn--lifehacer-1rb.com```

```Non-authoritative answer:
xn--lifehacer-1rb.com
	origin = dns1.yandex.net
	mail addr = iskalko.yandex.ru
	serial = 2016120703
	refresh = 14400
	retry = 900
	expire = 1209600
	minimum = 14400
xn--lifehacer-1rb.com	mail exchanger = 10 mx.yandex.net.
Name:	xn--lifehacer-1rb.com
Address: 78.110.60.230
xn--lifehacer-1rb.com	nameserver = dns2.yandex.net.
xn--lifehacer-1rb.com	text = "v=spf1 redirect=_spf.yandex.net"
xn--lifehacer-1rb.com	nameserver = dns1.yandex.net.
```

**Comment by drakofrost** [comment]
Nslookup works for me, but the address doesn't change in the browser, and also fails to resolve.

I'm using Fedora 25.

**Comment by drakofrost** [comment]
Nslookup works for me, but the address doesn't change when I paste it in the browser, and also doesn't resolve.

I'm using Fedora 25.

**Comment by mitchellkrogza** [comment]
Interesting one this, I think it's a browser problem.

Just did some tests as follows: 

**Ubuntu:**
Firefox and Chrome 
Paste address xn--lifehacer-1rb.com without http://
Opens page
nslookup - resolves

**MacOS Sierra:**
Firefox, Chrome, Safari 
Paste address xn--lifehacer-1rb.com without http://
Opens page
nslookup - resolves

**Windows XP:**
Firefox, Chrome
Paste address xn--lifehacer-1rb.com without http://
Opens page
nslookup - resolves

**Linux Mint:**
Firefox, Chrome 
Paste address xn--lifehacer-1rb.com without http://
Opens page
nslookup - resolves

**Android Phone:** (here's an issue)
Chrome
See screengrab
![android](https://cloud.githubusercontent.com/assets/9961541/23250349/5da742c0-f9b1-11e6-9bd5-7608e2d70c0a.jpg)

Android on Chrome incorrectly translates what I type into the search bar.

**Comment by mitchellkrogza** [comment]
Update: 

Sent an email from my desktop to my android phone with the links in the email. Clicking either one of the links in the email and they both open in chrome on Android. 

The fact that all nslookups resolve should be enough to satisfy that the hosts file will also understand the xn--- formats perfectly.

![email-android](https://cloud.githubusercontent.com/assets/9961541/23250613/8fde76ea-f9b2-11e6-9acb-ebab387463f2.jpg)

**Comment by PromoFaux** [comment]
Pi-hole dev chipping in here! @drakofrost I've just updated my `gravity.list` to include new changes to this list and I'm not hitting the same error with `lifehacĸer.com` causing `dnsmasq` to fall over... what version of `dnsmasq` do you have?

**Comment by PromoFaux** [comment]
@mitchellkrogza 
>lifehacĸer.com translates to xn--lifehacer-1rb.com

I notice you include both `xn--lifehacer-1rb.com` and `lifehacĸer.com` in your list. If the latter just redirects to the former, then `lifehacĸer.com` is superfluous, surely?

**Comment by mitchellkrogza** [comment]
Thanks, fixed in https://github.com/mitchellkrogza/Badd-Boyz-Hosts/commit/5001714c984350c67671e9c1e6806e73328a3126

**Comment by StevenBlack** [comment]
@mitchellkrogza @PromoDaux @drakofrost thank you, everyone, for your diligence!

I just pushed the latest sources to master.

**Comment by drakofrost** [comment]
@mitchellkrogza It may have to do with the fact that I'm running Pi-Hole as a DNS server, maybe Pi-Hole is blocking those addresses.

**Comment by drakofrost** [comment]
@PromoFaux dnsmasq version 2.66, I'm running CentOS 7 on DigitalOcean.

**Comment by PromoFaux** [comment]
>dnsmasq version 2.66,
@drakofrost Possibly something to do with it. I know debian repositories have version 2.72, and I have no issues with that, even with the special chars.

Latest version is 2.76, which you can compile from source.

**Comment by drakofrost** [comment]
@PromoFaux I left the process of compiling stuff in the 90's :)

**Comment by Atavic** [comment]
Go to Yandex.ru
Search for: lifehacĸer.com
1st result is there.

The ĸ in lifehacĸer is a letter of the Cyrillic script.

**Comment by funilrys** [comment]
This one should be also fixed by #520 in future updates @StevenBlack :+1:

**Comment by StevenBlack** [comment]
Thanks @funilrys.  Closing.
